src/main/java/edu/cornell/kfs/fp/document/RecurringDisbursementVoucherDocument.java

```diff
@@ -0,0 +1,138 @@
+                        break;
+                    }
+                    KualiDecimal partialAmount = datesAndAmounts.get(revDate);
+                    GeneralLedgerPendingEntry explicitPartialEntry = new GeneralLedgerPendingEntry();
+                    getGeneralLedgerPendingEntryService().populateExplicitGeneralLedgerPendingEntry(this, glpeSourceDetail, sequenceHelper, explicitPartialEntry);
+                    explicitPartialEntry.setFinancialDocumentReversalDate(revDate);
+                    explicitPartialEntry.setTransactionLedgerEntryAmount(isErrorCorrection ? partialAmount.negated() : partialAmount);
+                    customizeExplicitGeneralLedgerPendingEntry(glpeSourceDetail, explicitPartialEntry);
+                    addPendingEntry(explicitPartialEntry);
+                    sequenceHelper.increment();
+                    GeneralLedgerPendingEntry offsetEntry = new GeneralLedgerPendingEntry(explicitPartialEntry);
+                    processOffsetGeneralLedgerPendingEntry(sequenceHelper, glpeSourceDetail, explicitPartialEntry, offsetEntry);
+                    sequenceHelper.increment();
+                }
+                return;
+            }
+        }
+        getGeneralLedgerPendingEntryService().populateExplicitGeneralLedgerPendingEntry(this, glpeSourceDetail, sequenceHelper, explicitEntry);
+        customizeExplicitGeneralLedgerPendingEntry(glpeSourceDetail, explicitEntry);
+        addPendingEntry(explicitEntry);
+        sequenceHelper.increment();
+        GeneralLedgerPendingEntry offsetEntry = new GeneralLedgerPendingEntry(explicitEntry);
+        processOffsetGeneralLedgerPendingEntry(sequenceHelper, glpeSourceDetail, explicitEntry, offsetEntry);
+    }
+	
+	@Override
+    public boolean generateGeneralLedgerPendingEntries(GeneralLedgerPendingEntrySourceDetail glpeSourceDetail,
+        GeneralLedgerPendingEntrySequenceHelper sequenceHelper) {
+        GeneralLedgerPendingEntry explicitEntry = new GeneralLedgerPendingEntry();
+        processExplicitGeneralLedgerPendingEntry(sequenceHelper, glpeSourceDetail, explicitEntry);
+        sequenceHelper.increment();
+        return true;
+    }
+	
+	@Override
+    protected boolean processOffsetGeneralLedgerPendingEntry(GeneralLedgerPendingEntrySequenceHelper sequenceHelper,
+        GeneralLedgerPendingEntrySourceDetail postable, GeneralLedgerPendingEntry explicitEntry, GeneralLedgerPendingEntry offsetEntry) {
+        boolean success = getGeneralLedgerPendingEntryService().populateOffsetGeneralLedgerPendingEntry(getPostingYear(), explicitEntry, sequenceHelper, offsetEntry);
+        success &= customizeOffsetGeneralLedgerPendingEntry(postable, explicitEntry, offsetEntry);
+        addPendingEntry(offsetEntry);
+        return success;
+    }
+	
+	protected HomeOriginationService getHomeOriginationService() {
+		return SpringContext.getBean(HomeOriginationService.class);
+	}
+	
+	protected GeneralLedgerPendingEntryService getGeneralLedgerPendingEntryService() {
```

Comment:
should also probably be refactored to be more like getScheduledAccountingLineService() and add a setter to allow for more easy testing